yt/frontends/ramses/field_handlers.py

```diff
@@ -19,49 +20,26 @@ def register_field_handler(ph):
     FIELD_HANDLERS.add(ph)
 
 
-PRESENT_FIELD_FILES = {}
 DETECTED_FIELDS = {}
 
 
-class RAMSESFieldFileHandlerRegistry(type):
+class RAMSESFieldFileHandlerRegistry(abc.ABCMeta):
     """
     This is a base class that on instantiation registers the file
     handler into the list. Used as a metaclass.
     """
 
     def __new__(meta, name, bases, class_dict):
-        cls = type.__new__(meta, name, bases, class_dict)
+        cls = abc.ABCMeta.__new__(meta, name, bases, class_dict)
         if cls.ftype is not None:
             register_field_handler(cls)
-        return cls
-
-
-class FieldFileHandler(metaclass=RAMSESFieldFileHandlerRegistry):
-    """
-    Abstract class to handle particles in RAMSES. Each instance
-    represents a single file (one domain).
-
-    To add support to a new particle file, inherit from this class and
-    implement all functions containing a `NotImplementedError`.
-
-    See `SinkParticleFileHandler` for an example implementation."""
-
-    # These properties are static properties
-    ftype = None  # The name to give to the field type
-    fname = None  # The name of the file(s)
-    attrs = None  # The attributes of the header
-    known_fields = None  # A list of tuple containing the field name and its type
-    config_field = None  # Name of the config section (if any)
 
-    file_descriptor = None  # The name of the file descriptor (if any)
+        cls._unique_registry = {}
+        return cls
 
-    # These properties are computed dynamically
-    field_offsets = None  # Mapping from field to offset in file
-    field_types = (
-        None  # Mapping from field to the type of the data (float, integer, ...)
-    )
 
-    def __init__(self, domain):
+class HandlerMixin:
```

Comment:
It may be worth adding a docstring to this class since it's not really obvious what its purpose is at first glance.